public/app/features/alerting/unified/state/AlertmanagerContext.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+
+import { useQueryParams } from 'app/core/hooks/useQueryParams';
+import store from 'app/core/store';
+
+import { useAlertManagersByPermission } from '../hooks/useAlertManagerSources';
+import { ALERTMANAGER_NAME_LOCAL_STORAGE_KEY, ALERTMANAGER_NAME_QUERY_KEY } from '../utils/constants';
+import { AlertManagerDataSource, GRAFANA_RULES_SOURCE_NAME } from '../utils/datasource';
+
+interface Context {
+  permissions: 'instance' | 'notification';
+  selectedAlertmanager: string | undefined;
+  availableAlertManagers: AlertManagerDataSource[];
+  setSelectedAlertmanager: (name: string) => void;
+  setPermissions: (accessType: Context['permissions']) => void;
+}
+
+const SelectedAlertmanagerContext = React.createContext<Context | undefined>(undefined);
+
+// TODO don't use setPermissions but pass permissions down to the alertmanager provider via props
+const SelectedAlertmanagerProvider = ({ children }: React.PropsWithChildren) => {
+  const [permissions, setPermissions] = React.useState<Context['permissions']>('instance');
+  const [queryParams, updateQueryParams] = useQueryParams();
+  const availableAlertManagers = useAlertManagersByPermission(permissions);
+
+  const updateSelectedAlertmanager = React.useCallback(
+    (selectedAlertManager: string) => {
+      if (!isAlertManagerAvailable(availableAlertManagers, selectedAlertManager)) {
+        return;
+      }
+
+      if (selectedAlertManager === GRAFANA_RULES_SOURCE_NAME) {
+        store.delete(ALERTMANAGER_NAME_LOCAL_STORAGE_KEY);
+        updateQueryParams({ [ALERTMANAGER_NAME_QUERY_KEY]: null });
```

Comment:
why do we do this?

Reply:
This code had been around before but I assume it's to

a. Set the selected AM as a query param so links can be shared with others while maintaining the AM where we have to find the resource (mute timing, silence, etc)
b. Local storage is used for maintaining the selected AM between browser sessions or full refreshes of the page.

Reply:
Oh sorry, if the question was specifically about setting it to "null" I had the exact same question and I have no idea :)

Reply:
It seems we only add the alertmanager query param if it's not Grafana